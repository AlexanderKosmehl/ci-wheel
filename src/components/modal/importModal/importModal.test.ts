import {
  describe, expect, it, jest,
} from '@jest/globals';
import generateImportModal from './importModal';

describe('generateImportModal', () => {
  const onClose = jest.fn(() => {});
  const onImport = jest.fn((_importText: string) => {});

  const importModal = generateImportModal({
    onClose,
    onImport,
  });

  const closeButton = importModal.querySelector<HTMLButtonElement>('.closeIcon');

  const textArea = importModal.querySelector<HTMLTextAreaElement>('textarea');
  const importButton = importModal.querySelector<HTMLButtonElement>('.importButton');

  it('generates component correctly', () => {
    expect(importModal).toMatchSnapshot();
  });

  it('adds working onClose callback', () => {
    if (!closeButton) throw Error('No closeButton rendered!');

    closeButton.click();
    expect(onClose).toBeCalled();
  });

  it('adds working onImport callback', () => {
    if (!textArea || !importButton) throw Error('No textarea or importButton rendered!');

    const testvalue = 'Test';

    textArea.value = testvalue;
    importButton.click();

    expect(onImport).toBeCalled();
    expect(onImport).toBeCalledWith(testvalue);
  });
});
