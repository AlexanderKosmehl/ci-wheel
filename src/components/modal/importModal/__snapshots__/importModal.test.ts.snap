// Vitest Snapshot v1

exports[`generateImportModal > generates component correctly 1`] = `
<div
  class="_container_0360d7"
>
  <div
    class="_content_0360d7"
  >
    <div
      class="_header_0360d7"
    >
      <h2
        class="_headerText_0360d7"
      >
        Import
      </h2>
      <button
        class="_button_7c1f96 _closeButton_0360d7"
        data-test="modalCloseButton"
      >
        <img
          class="_icon_7c1f96"
          src="/src/assets/close-icon.svg"
        />
      </button>
    </div>
    <div
      class="_importModalContainer_e2444d"
    >
      <textarea
        class="_importTextarea_e2444d"
        data-test="modalTextarea"
        placeholder="Neue Einträge zeilenweise einfügen..."
      />
      <div
        class="_importButtonWrapper_e2444d"
      >
        <button
          class="_importButton_e2444d"
          data-test="modalImportButton"
        >
          Importieren
        </button>
      </div>
    </div>
  </div>
</div>
`;
